File: python/easy/217.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        tmp = set()
        for i in nums:
            if i in tmp:
                return True
            tmp.add(i)
        return False


solve = Solution()

print(solve.containsDuplicate(nums))

File: 217.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> hash_table;
        for (auto& num : nums) {
            if (hash_table.find(num) != hash_table.end()) {
                return true;
            }
            hash_table.insert(num);
        }
        return false;
    }
};
